config/config.go

```diff
@@ -204,7 +206,25 @@ func (c *Config) Authenticate(r *http.Request) error {
 	if err != nil {
 		return err
 	}
-	return c.auth(r)
+	return c.credentialsProvider.SetHeaders(r)
+}
+
+// Authenticate returns an OAuth token for the current configuration.
+// It will return an error if the CredentialsStrategy does not support OAuth tokens.
+func (c *Config) GetToken() (*oauth2.Token, error) {
+	err := c.EnsureResolved()
+	if err != nil {
+		return nil, err
+	}
+	err = c.authenticateIfNeeded()
+	if err != nil {
+		return nil, err
+	}
+	if h, ok := c.credentialsProvider.(credentials.OAuthCredentialsProvider); ok {
+		return h.Token()
+	} else {
+		return nil, fmt.Errorf("OAuth Token can only be retrieved for OauthCredentialsProvider")
```

Comment:
Slightly more useful error message for users
```suggestion
		return nil, fmt.Errorf("OAuth Token not supported for current auth type %s", c.AuthType)
```